File: Benchmarking/30 Euler/euler21/euler21.cpp
//
//  main.cpp
//  Euler21
//
//  Created by Sonni on 20/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>

unsigned int sum_divisors(unsigned int n)
{
    unsigned int sum = 0;
    int k = n - 1;

    for (int i = 1; i <= k; i++)
    {
        if (n % i == 0)
            sum += i;
    }
    return sum;
}

unsigned int amica(unsigned int a)
{
    unsigned int amicable = 0;
    unsigned int sum = 0;
    
    for (unsigned int i = 1; i <= a; i++)
    {
        unsigned int b = sum_divisors(i);
        
        unsigned int c = sum_divisors(b);
        if (i == c && i != b)
        {
            amicable++;
            sum += i;
        }
    }
    return sum;
}

int main(int argc, const char * argv[]) {
    unsigned int limit = atoi(argv[1]);
    std::cout << amica(limit) << std::endl;
    return 0;
}

File: Benchmarking/30 Euler/euler21/euler21.py
import sys

limit = int(sys.argv[1])
def sum_divisors(n):
  sum = 0
  k = n-1
  for i in range(1, k+1):
    if n % i == 0:
      sum += i
  return sum

def amica(a):
  amicable = 0
  sum = 0
  for i in range(1, a+1):
    b = sum_divisors(i)
    c = sum_divisors(b)
    if i == c and i != b:
      amicable += 1
      sum += i
  return sum

print amica(limit)
